ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -540,24 +551,31 @@ private int HandlePermEvent(ref ProjFS.Event ev)
                 case NotificationType.PreDelete:
                     return this.OnPreDelete(relativePath, isDirectory);
 
-                case NotificationType.FileModified:
-                    this.OnFileModified(relativePath);
-                    return Result.Success;
+                case NotificationType.PreRename:
+                    return this.OnPreRename(relativePath, relativeDestinationPath, isDirectory);
```

Comment:
What will `relativePath` and `relativeDestinationPath` be when the rename crosses the virtualization root boundary?

Reply:
Good question!  Fortunately, that can't happen, because `rename(2)` won't work across device boundaries, i.e., between mounted filesystems, and the whole repository/working directory is a separate mounted filesystem.

As it happens, I just rewrote some functional tests with this in mind -- see 6eb3597ece6cfb436a2478a5a50771fe5affbf69, which uses `BashRunner` instead of `SystemIORunner` on Linux when executing tests which move things in/out of the repository.  That's because command-line `mv` will helpfully fall back to a copy-and-delete strategy.  But an actual `rename()` file operation will error out before it gets to the point that we'd need to send any notifications.